src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/snippet/ParseException.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2020, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.javadoc.internal.doclets.toolkit.taglets.snippet;
+
+/**
+ * An exception thrown by {@link MarkupParser}.
+ *
+ * <p><b>This is NOT part of any supported API.
+ * If you write code that depends on this, you do so at your own risk.
+ * This code and its internal interfaces are subject to change or
+ * deletion without notice.</b>
+ */
+public class ParseException extends Exception {
+
+    @java.io.Serial
+    private static final long serialVersionUID = 1;
+
+    private final int index;
+
+    /**
+     * Constructs an exception with a message and an approximate position of
+     * the erroneous portion of markup within a parsed line.
+     *
+     * @param message the description
+     * @param position the approximate position
+     */
+    public ParseException(String message, int position) {
```

Comment:
I know it is common to use English-only strings in exceptions that are not intended to be seen in normal use by an end user, but here, you're using `ParseException` to contain content that is intended to appear in end-user error messages, triggered by errors in the content provided by an end-user.   As such, it is close to a firm requirement that these messages can be localized.

Reply:
One possible solution at this time that would be minimally invasive would be to replace `new ParseException(String message, int pos)` with use of a method that localizes the string first, as in,
```
ParseException newParseException(int pos, String resourceKey, Object... args)
```
so the code becomes something like
```
   throw newParseException(pos, key, args);
```

Reply:
> I know it is common to use English-only strings in exceptions that are not intended to be seen in normal use by an end user, but here, you're using `ParseException` to contain content that is intended to appear in end-user error messages, triggered by errors in the content provided by an end-user. As such, it is close to a firm requirement that these messages can be localized.

You are completely right. In fact, I was going to internationalize those messages initially in this PR, but then decided to postpone it until the https://bugs.openjdk.java.net/browse/JDK-8273544 follow-up bug. I removed the respective FIXMEs in one of the updates either to this PR or the originating sandbox branch.

One reason I was postponing internationalization is that I was sure neither about the wording nor about the granularity of the messages. But I guess, it's not that big a deal to amend either of those later.